cpp/src/arrow/array/array_binary_test.cc

```diff
@@ -1007,11 +1013,97 @@ class TestBaseBinaryDataVisitor : public ::testing::Test {
  protected:
   std::shared_ptr<DataType> type_;
 };
-
 TYPED_TEST_SUITE(TestBaseBinaryDataVisitor, BaseBinaryOrBinaryViewLikeArrowTypes);
 
 TYPED_TEST(TestBaseBinaryDataVisitor, Basics) { this->TestBasics(); }
 
 TYPED_TEST(TestBaseBinaryDataVisitor, Sliced) { this->TestSliced(); }
 
+template <typename Type>
+class TestCompactArray : public ::testing::Test {
+  using ArrayType = typename TypeTraits<Type>::ArrayType;
+
+ public:
+  void SetUp() override { type_ = TypeTraits<Type>::type_singleton(); }
+  void TestType() {
```

Comment:
Nit: can you leave empty lines between methods?
```suggestion

  void TestType() {
```